src/xcode/ENA/ENA/Source/Scenes/Events/OnBehalfCheckinSubmission/OnBehalfCheckinSubmissionCoordinator.swift

```diff
@@ -0,0 +1,244 @@
+	}
+	
+	// MARK: - Private
+
+	private weak var parentViewController: UIViewController!
+	private var navigationController: DismissHandlingNavigationController!
+
+	private let appConfiguration: AppConfigurationProviding
+	private let eventStore: EventStoringProviding
+	private let client: Client
+
+	private var traceLocationSelectionViewModel: OnBehalfTraceLocationSelectionViewModel?
+
+	// MARK: Show Screens
+
+	private lazy var infoScreen: UIViewController = {
+		let infoViewController = OnBehalfInfoViewController(
+			onPrimaryButtonTap: { [weak self] in
+				self?.showTraceLocationSelectionScreen()
+			},
+			onDismiss: { [weak self] in
+				self?.parentViewController.dismiss(animated: true)
+			}
+		)
+
+		let footerViewController = FooterViewController(
+			FooterViewModel(
+				primaryButtonName: AppStrings.OnBehalfCheckinSubmission.Info.primaryButtonTitle,
+				isPrimaryButtonEnabled: true,
+				isSecondaryButtonHidden: true,
+				backgroundColor: .enaColor(for: .background)
+			)
+		)
+
+		return TopBottomContainerViewController(
+			topController: infoViewController,
+			bottomController: footerViewController
+		)
+	}()
+
+	private func showTraceLocationSelectionScreen() {
+		let viewModel = OnBehalfTraceLocationSelectionViewModel(traceLocations: eventStore.traceLocationsPublisher.value)
+		traceLocationSelectionViewModel = viewModel
+
+		let traceLocationSelectionViewController = OnBehalfTraceLocationSelectionViewController(
+			viewModel: viewModel,
+			onScanQRCodeCellTap: { [weak self] in
+				self?.showQRCodeScanner()
+			},
+			onMissingPermissionsButtonTap: { [weak self] in
+				self?.showSettings()
+			},
+			onPrimaryButtonTap: { [weak self] selectedTraceLocation in
+				self?.showDateTimeSelectionSelectionScreen(
+					traceLocation: selectedTraceLocation
+				)
+			},
+			onDismiss: { [weak self] in
+				self?.parentViewController.dismiss(animated: true)
+			}
+		)
+
+		let footerViewController = FooterViewController(
+			FooterViewModel(
+				primaryButtonName: AppStrings.OnBehalfCheckinSubmission.TraceLocationSelection.primaryButtonTitle,
+				isPrimaryButtonEnabled: true,
+				isSecondaryButtonHidden: true,
+				backgroundColor: .enaColor(for: .background)
+			)
+		)
+
+		let containerViewController = TopBottomContainerViewController(
+			topController: traceLocationSelectionViewController,
+			bottomController: footerViewController
+		)
+
+		navigationController.pushViewController(containerViewController, animated: true)
+	}
+
+	private func showQRCodeScanner() {
+		let qrCodeScanner = CheckinQRCodeScannerViewController(
+			qrCodeVerificationHelper: QRCodeVerificationHelper(),
+			appConfiguration: appConfiguration,
+			didScanCheckin: { [weak self] traceLocation in
+				self?.navigationController.dismiss(animated: true) {
+					self?.showDateTimeSelectionSelectionScreen(traceLocation: traceLocation)
+				}
+			},
+			dismiss: { [weak self] in
+				self?.traceLocationSelectionViewModel?.updateForCameraPermission()
```

Comment:
Would be a lot easier to understand if the logic would not route through the viewmodel, over a published property.
But just calling a reload on the view controller here.

Reply:
Makes sense, I am reloading directly on the view controller now instead of going through the model